### PR TITLE
 Support property to define tsconfig file

### LINKED ISSUE #1310: Support property to define tsconfig file
We should support property `sonar.typescript.tsconfigPath` to define tsconfig file location by the user same way as we did in SonarTS. This should have priority before automatic detection.

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #1310

### CODE REVIEW

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java [hunk 1]
```diff
@@ -160,6 +162,36 @@ public void should_lookup_tsconfig_files() throws Exception {
...
+    typeScriptSensor.execute(ctx);
+
+    ArgumentCaptor<TypeScriptAnalysisRequest> request = ArgumentCaptor.forClass(TypeScriptAnalysisRequest.class);
+    verify(eslintBridgeServerMock).analyzeTypeScript(request.capture());
+
+    assertThat(request.getValue().tsConfigs).containsOnly(baseDir.resolve("custom.tsconfig.json").toAbsolutePath().toString());
+  }
+
+  @Test
+  public void should_fail_with_wrong_tsconfig() throws Exception {
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java]: I would rename test to `should_log_and_stop_with_wrong_tsconfig`

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/TypeScriptAnalysisTest.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public void test() throws Exception {
     assertThat(issuesList).hasSize(1);
     assertThat(issuesList.get(0).getLine()).isEqualTo(2);
 
-    assertThat(result.getLogsLines(log -> log.contains("Found 3 tsconfig.json files"))).hasSize(1);
+    assertThat(result.getLogsLines(log -> log.contains("Found 1 tsconfig.json files"))).hasSize(1);
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/TypeScriptAnalysisTest.java]: I think we should make it `file(s)`

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/TypeScriptSensor.java [hunk 3]
```diff
@@ -111,18 +118,42 @@ private static void saveHighlights(InputFile file, SensorContext context, Analys
...
-        tsconfigs = files
-          .filter(p -> p.endsWith("tsconfig.json"))
-          .map(p -> p.toAbsolutePath().toString())
-          .collect(Collectors.toList());
+      Optional<String> tsConfigProperty = context.config().get(JavaScriptPlugin.TSCONFIG_PATH);
+      if (tsConfigProperty.isPresent()) {
+        Path tsconfig = Paths.get(tsConfigProperty.get());
+        tsconfig = tsconfig.isAbsolute() ? tsconfig : context.fileSystem().baseDir().toPath().resolve(tsconfig);
+        if (!tsconfig.toFile().exists()) {
+          String msg = format("Provided tsconfig.json path doesn't exists. Path: '%s'", tsconfig);
```
Comment (yassin-kammoun-sonarsource) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/TypeScriptSensor.java]: Typo in `exists`.

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java [hunk 4]
```diff
@@ -160,6 +163,36 @@ public void should_lookup_tsconfig_files() throws Exception {
...
+    Path baseDir = tempFolder.newDir().toPath();
+    SensorContextTester ctx = SensorContextTester.create(baseDir);
+    ctx.setSettings(new MapSettings().setProperty("sonar.typescript.tsconfigPath", "wrong.json"));
+    createInputFile(ctx);
+    TypeScriptSensor typeScriptSensor = createSensor();
+    typeScriptSensor.execute(ctx);
+
+    verify(eslintBridgeServerMock, never()).analyzeTypeScript(any());
+
+    assertThat(logTester.logs(LoggerLevel.ERROR)).contains("Provided tsconfig.json path doesn't exists. Path: '" + baseDir.resolve("wrong.json") + "'");
```
Comment (yassin-kammoun-sonarsource) [inline sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/TypeScriptSensorTest.java]: Same typo.

**Pull request merged**